pkg/controllers/provisioning/scheduling/topology.go

```diff
@@ -278,6 +278,28 @@ func (t *Topology) countDomains(ctx context.Context, tg *TopologyGroup) error {
 		pods = append(pods, podList.Items...)
 	}
 
+	// capture new domain values from existing nodes that may not have any pods selected by the topology group
+	// scheduled to them already
+	t.cluster.ForEachNode(func(n *state.StateNode) bool {
```

Comment:
Noting down that we have plans to move this over into domainGroups since domainGroups should really capture _all_ of the domains that are eligible across the cluster (including NodePools) rather than just the NodePools (which is what they are doing right now)